src/blocks/core/group.js

```diff
@@ -0,0 +1,203 @@
+/**
+ * Customizations for the core group block.
+ */
+
+import { createBlock } from '@wordpress/blocks';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { InspectorControls } from '@wordpress/block-editor';
+import { PanelBody, TextControl, ToggleControl } from '@wordpress/components';
+import { addFilter, removeFilter } from '@wordpress/hooks';
+import { __ } from '@wordpress/i18n';
+
+const toIdString = ( string ) => {
+	return string
+		.toLowerCase()
+		.replace( /[^a-z0-9]/g, ' ' )
+		.trim()
+		.split( /\s+/ )
+		.join( '-' );
+};
```

Comment:
If possible, I think it might be nice to use one of the core functions for generating slugs, like this one:

https://github.com/WordPress/gutenberg/blob/9dc6483e4280db4aa716f5a55bf393d4a430ef99/packages/block-library/src/heading/autogenerate-anchors.js#L33-L44

But this works well enough for latin-ext text, so no blockers.